### ISSUE TITLE
Problem with signing into Firefox Account launched into Edit Account screen

### ISSUE THREAD

**Issue by Dan884** [mozilla-mobile/fenix#16232]
It opens a window to change account picture, display name, etc without taking you back to the normal screen to let you keep setting up new install. I couldn't figure out how to get out of this window without logging out. 

## Steps to reproduce
Click Turn on sync. Click Use email instead. Enter username, password, Auth code. 

### Expected behavior
Sign you in and take you back to initial install screen. 

### Actual behavior
See description above

### Device information
Samsung S20 FE 5G

* Android device: Android 10
* Fenix version: ?
 Nightly 201027 17:01 (Build #2015772171)

**Comment by Dan884** [comment]
More information. When I intially clicked on "Use email instead" , the sign in screen said "Sign in" . On next line it said "Continue to Firefox Sync" . Then I left the Firefox app to copy the username. Then when I went back to Firefox, it had gone back to the previous screen and I had to click "Use email instead" . Now the screen says "Sign in" . But the next line says "Continue to Account Settings" . So the problem only occurs if you leave Firefox when it is asking for username and then go back into Firefox.

**Comment by sv-ohorvath** [comment]
I encountered this issue as well on BEta 83.0.0-beta3 and Nightly 11/4. It happens while signing in through the Sign in with email flow and also on the Create account flow. 
It is easier for me to reproduce it on the Create a new account flow:
1. Go to Turn on Sync>Sing in using any method, but save the credentials when asked.
2. Sign out.
3. Go back to sync, now tap the "Create one" link to go to the new account flow. 
4. As you have an account already saved, it will auto-fill the username/email.
5. Continue and sign in with the saved account.

After signing in, you will see the Manage account page. If you close the page by pressing the "x" button in the custom tab toolbar, it will exit back to Settings and the account won't be signed in.

**Comment by st3fan** [comment]
@grigoryk would you mind taking a peek?

**Comment by sv-ohorvath** [comment]
Can also reproduce when the user was previously signed in, then signed out so the credentials are saved. 
Going to Sing in using email, with the credentials already pre-filled, you only tap Sign in and you are taken to the Manage account screen. 
When you leave that screen you are not logged in.

*The difference between this and my previous comment is that you don't have to go through "Use a different account".

**Comment by st3fan** [comment]
Log posted by Janet:

```
11-13 11:19:53.263 31601 18610 W FirefoxAccountStateMachine: Got invalid event 'mozilla.components.service.fxa.manager.Event$Account$Start@c743f3e' for state Active(progressState=BeginningAuthentication).
11-13 11:20:44.076 31601 31601 W mozac-fxawebchannel: Unrecognized WebChannel command: fxaccounts:login
11-13 11:20:44.077 31601 31601 E mozac-fxawebchannel: Error while processing WebChannel command
11-13 11:20:44.077 31601 31601 E mozac-fxawebchannel: org.json.JSONException: Couldn't get WebChannel command
11-13 11:20:44.077 31601 31601 E mozac-fxawebchannel: 	at mozilla.components.feature.accounts.FxaWebChannelFeature$WebChannelViewContentMessageHandler.onPortMessage(FxaWebChannelFeature.kt:65)
11-13 11:20:44.077 31601 31601 E mozac-fxawebchannel: 	at mozilla.components.browser.engine.gecko.webextension.GeckoWebExtension$registerContentMessageHandler$portDelegate$1.onPortMessage(GeckoWebExtension.kt:1)
11-13 11:20:44.077 31601 31601 E mozac-fxawebchannel: 	at org.mozilla.geckoview.WebExtension$Port.portMessage(WebExtension.java:2)
11-13 11:20:44.077 31601 31601 E mozac-fxawebchannel: 	at org.mozilla.geckoview.WebExtension$Port.access$100(WebExtension.java:1)
11-13 11:20:44.077 31601 31601 E mozac-fxawebchannel: 	at org.mozilla.geckoview.WebExtension$Port$1.handleMessage(WebExtension.java:4)
11-13 11:20:44.077 31601 31601 E mozac-fxawebchannel: 	at org.mozilla.gecko.EventDispatcher$2.run(EventDispatcher.java:1)
11-13 11:20:44.077 31601 31601 E mozac-fxawebchannel: 	at android.os.Handler.handleCallback(Handler.java:883)
11-13 11:20:44.077 31601 31601 E mozac-fxawebchannel: 	at android.os.Handler.dispatchMessage(Handler.java:100)
11-13 11:20:44.077 31601 31601 E mozac-fxawebchannel: 	at android.os.Looper.loop(Looper.java:237)
11-13 11:20:44.077 31601 31601 E mozac-fxawebchannel: 	at android.app.ActivityThread.main(ActivityThread.java:8107)
11-13 11:20:44.077 31601 31601 E mozac-fxawebchannel: 	at java.lang.reflect.Method.invoke(Native Method)
11-13 11:20:44.077 31601 31601 E mozac-fxawebchannel: 	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:496)
11-13 11:20:44.077 31601 31601 E mozac-fxawebchannel: 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1100)
11-13 11:21:34.546 31601 18610 W FirefoxAccountStateMachine: Got invalid event 'mozilla.components.service.fxa.manager.Event$Account$Start@c743f3e' for state Active(progressState=BeginningAuthentication).
11-13 11:22:42.731 31601 18610 W FirefoxAccountStateMachine: Got invalid event 'mozilla.components.service.fxa.manager.Event$Account$BeginEmailFlow@5b8f45f' for state Active(progressState=BeginningAuthentication).
11-13 11:22:42.979 31601 18610 W FirefoxAccountStateMachine: Got invalid event 'mozilla.components.service.fxa.manager.Event$Account$Start@c743f3e' for state Active(progressState=BeginningAuthentication).
```

**Comment by grigoryk** [comment]
I can't seem to be able to reproduce this locally.

However, from Janet's logs:
`Unrecognized WebChannel command: fxaccounts:login`

Is the clue. We're waiting for the `fxaccounts:oauth_login` command instead, and indeed `fxaccounts:login` will be treated as unrecognized.

**Comment by grigoryk** [comment]
Okay, we were able to reproduce it and fix (at least one obvious problem). See https://github.com/mozilla-mobile/android-components/pull/8976 - this should be in Nightly sometime on Monday.

Reliable STRs for the bug:
- begin authentication as normal
- cancel authentication without completing it (by pressing back, or closing the custom tab via X)
- try authentication again. at this point, it will no longer work - you'd end up on the Manage Account screen after completing
 -- restarting the app should fix this state though

**Comment by sv-ohorvath** [comment]
We've verified the fix on the 831.0-rc2 build and it is working as expected now. Found another small issue and logged it on FxA: https://github.com/mozilla/fxa/issues/6944
On Nightly 11/16 the fix hasn't landed yet and is still reproducing ( current AC version 66.0.20201109143146).
Closing this issue as fixed for now.
